File: src/Item38/java/ExtendedOperation.java
package Item38.java;

public enum ExtendedOperation implements Operation {

    EXP("^"){
        @Override
        public double apply(double x, double y) { return Math.pow(x, y); }
    },
    REMINDER("%"){
        @Override
        public double apply(double x, double y) { return x % y; }
    };

    private final String symbol;

    ExtendedOperation(String symbol){
        this.symbol = symbol;
    }

    @Override
    public String toString() {
        return symbol;
    }
}

File: src/Item38/kotlin/ExtendedOperation.kt
package Item38.kotlin

import Item38.java.Operation

enum class ExtendedOperation private constructor(private val symbol: String) : Operation {

    EXP("^") {
        override fun apply(x: Double, y: Double): Double {
            return Math.pow(x, y)
        }
    },
    REMINDER("%") {
        override fun apply(x: Double, y: Double): Double {
            return x % y
        }
    };

    override fun toString(): String {
        return symbol
    }
}